### ISSUE TITLE
flash-attention with -100 in variable positions

### ISSUE THREAD

**Issue by congchan** [Dao-AILab/flash-attention#851]
We use -100 as the masking target to avoid computing loss on certain positions. For example:
```
input:   100, 200,  300, 400
target:  100, -100, 100, -100
```

My best guess is that flash-attention is **compatible** with such loss, since flash-attention only handle attention operation.

I am not familiar with flash-attention implementation. Could anyone help to explain or verify my concerns?

**Comment by Avelina9X** [comment]
Neither Flash Attention nor normal attention actually computes loss. As long as you are using a loss function which correctly interprets your loss there should be zero difference between Flash Attention, torch SDPA, doing attention with manual matmuls, and doing attention with einsums.

Flash Attention only implements the attention computation and is agnostic to your loss function and the rest of your architecture or training pipeline as a whole.
